moped-editor/src/views/projects/projectView/ProjectNameComplex.js

```diff
@@ -0,0 +1,292 @@
+
+const useStyles = makeStyles((theme) => ({
+  editIcons: {
+    cursor: "pointer",
+    marginTop: "16px",
+    fontSize: "2rem",
+  },
+  projectNameEditField: {
+    "font-size": "1.5rem",
+    "font-weight": "bold",
+  },
+  projectSecondaryNameEditField: {
+    "font-size": "1.0rem",
+    "font-weight": "bold",
+    "padding-top": "0.95rem",
+  },
+}));
+
+const ProjectNameComplex = (props) => {
+  // Incoming props:
+    // projectData={data.moped_project[0]} ✅
+    // projectId={projectId} ✅
+    // editable={true} ❓
+    // isEditing={isEditing} ✅
+    // setIsEditing={setIsEditing} ✅
+    // updatedCallback={handleNameUpdate} ✅
+
+  const classes = useStyles();
+
+  // indicates that the primary title isn't valid via the error state on the input field
+  const [primaryTitleError, setPrimaryTitleError] = useState(false);
+
+  const DEFAULT_SNACKBAR_STATE = {
+      open: false,
+      message: null,
+      severity: "success",
+    };
+
+  // used to control the snackbar's contents and visibility
+  const [snackbarState, setSnackbarState] = useState(DEFAULT_SNACKBAR_STATE);
+
+  // apollo hook returning the async function to update the project names
+  const [updateProjectNames] = useMutation(UPDATE_PROJECT_NAMES_QUERY);
+
+
+
+  // this function is fired when the user clicks on the check mark to save the names
+  const handleAcceptClick = (e) => {
+    e.preventDefault();
+
+    const projectName = document.getElementById("project_name").value;
+    const secondaryName = document.getElementById("secondary_name").value;
+
+    // Check if the primary name is defined and has meaningful content
+    if (!projectName || projectName.trim() === "") { // if it does not, set the error state
+      setPrimaryTitleError(true);
+    } else { // and if it passes the test, clear any error state and save it via a mutation
+      setPrimaryTitleError(false);
+      updateProjectNames({
+        variables: {
+          projectId: props.projectId,
+          projectName: projectName,
+          projectNameSecondary: secondaryName,
+        },
+      })
+      .then((res) => { // if the mutation is successful, show a success snackbar
+        setSnackbarState({
+          open: true,
+          message: <span>Success! The project names have been updated!</span>,
+          severity: "success",
+        });
+      }) 
+      .catch((error) => { // and if it fails, show an error snackbar
+        setSnackbarState({
+          open: true,
+          message: <span>Error. The project names have not been updated.</span>,
+          severity: "error",
+        });
+        })
+        .finally(() => { // return to the view mode, clear the snackbar after 3 seconds, 
+                         // and alert the parent component of the change
+        props.setIsEditing(false);
+        setTimeout(() => setSnackbarState(DEFAULT_SNACKBAR_STATE), 3000);
+        props.updatedCallback();
+      });
+    }
+  };
```

Comment:
can you rerun prettier on this file? the spacing threw me off on when the finally was happening

Reply:
Absolutely - thank you for asking. I just got that installed on my linux machine, which I should have done 1st thing. https://github.com/cityofaustin/atd-moped/pull/1302/commits/e3eab73343446801fade17868ce9083bbd46f509